File: Ex6.java
package Ex6;

class Motor {
    private String tipo; 
    private int potencia; 

    public Motor(String tipo, int potencia) {
        this.tipo = tipo;
        this.potencia = potencia;
    }

    public String getTipo() {
        return tipo;
    }

    public int getPotencia() {
        return potencia;
    }

    public void exibirInformacoes() {
        System.out.println("Motor tipo: " + tipo + ", Potência: " + potencia + " cv");
    }
}

class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private Motor motor;

    public Carro(String marca, String modelo, int ano, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.motor = motor;
    }

    public void exibirDetalhes() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano);
        motor.exibirInformacoes(); 
    }

    public static void main(String[] args) {
        Motor motor1 = new Motor("Gasolina", 150);
        Carro carro1 = new Carro("Toyota", "Corolla", 2020, motor1);

        Motor motor2 = new Motor("Elétrico", 200);
        Carro carro2 = new Carro("Tesla", "Model 3", 2022, motor2);

        carro1.exibirDetalhes();
        carro2.exibirDetalhes();
    }
}

File: ex6.go
package main

import (
	"fmt"
)

type Motor struct {
	Modelo   string
	Potencia int
}

type Carro struct {
	Marca string
	Modelo string
	Ano int
	Motor Motor
}

func (c Carro) ExibirDetalhes() {
	fmt.Printf("Marca: %s, Modelo: %s, Ano: %d, Motor: %s, Potência: %d HP\n",
		c.Marca, c.Modelo, c.Ano, c.Motor.Modelo, c.Motor.Potencia)
}

func main() {
	motor1 := Motor{Modelo: "V8", Potencia: 400}
	carro1 := Carro{Marca: "Ford", Modelo: "Mustang", Ano: 2021, Motor: motor1}

	motor2 := Motor{Modelo: "V6", Potencia: 300}
	carro2 := Carro{Marca: "Chevrolet", Modelo: "Camaro", Ano: 2022, Motor: motor2}

	carro1.ExibirDetalhes()
	carro2.ExibirDetalhes()
}
